public/app/features/search/components/ManageDashboards.test.tsx

```diff
@@ -0,0 +1,63 @@
+import { render, screen, waitFor } from '@testing-library/react';
+import React from 'react';
+import { Provider } from 'react-redux';
+
+import { contextSrv } from 'app/core/services/context_srv';
+import { configureStore } from 'app/store/configureStore';
+import { FolderDTO } from 'app/types';
+
+import ManageDashboardsNew from './ManageDashboardsNew';
+
+jest.mock('app/core/services/context_srv', () => {
+  const originMock = jest.requireActual('app/core/services/context_srv');
+
+  return {
+    ...originMock,
+    contextSrv: {
+      ...originMock.context_srv,
+      user: {},
+      hasAccess: jest.fn(() => false),
+    },
+  };
+});
+
+const setup = async (options?: { folder?: FolderDTO }) => {
+  const { folder = {} as FolderDTO } = options || {};
+  const store = configureStore();
+
+  const { rerender } = await waitFor(() =>
+    render(
+      <Provider store={store}>
+        <ManageDashboardsNew folder={folder} />
+      </Provider>
+    )
+  );
+
+  return { rerender, store };
+};
+
+jest.spyOn(console, 'error').mockImplementation();
+
+describe('ManageDashboards', () => {
+  beforeEach(() => {
+    (contextSrv.hasAccess as jest.Mock).mockClear();
+  });
+  it('should show dashboard actions if user has correct permissions', async () => {
+    (contextSrv.hasAccess as jest.Mock).mockReturnValue(false);
+
+    const { rerender, store } = await setup();
+
+    expect(screen.queryByRole('button', { name: /new/i })).not.toBeInTheDocument();
```

Comment:
Nitpicky but I didn't notice that this one test was both testing for having permissions and not having permissions, should we separate this into 2 tests so its easier to understand what each of them does? Or alternatively change the title to reflect that it is testing both scenarios